include/deal.II/base/iterator_range.h

```diff
@@ -172,7 +172,31 @@ class IteratorRange
      * object than the iterator represented by the argument.
      */
     bool
-    operator!=(const IteratorOverIterators &i_o_i);
+    operator!=(const IteratorOverIterators &i_o_i) const;
+
+    /**
+     * Implicit conversion operator.
+     *
+     * @warning After conversion from an IteratorOverIterators to the
+     * BaseIterator, it can no longer be guarenteed that iteration on
+     * the returned object results in an iterator to a valid object.
```

Comment:
I don't understand what this sentence means. If this isn't working, why does it make sense to allow implicit conversion?

Reply:
My comment was related to the validity of subsequent iteration of the returned (and converted) iterator. I'm worried about this:
```c++
IteratorRange<...> it_range (...);

IteratorOverIterators ioi = it_range.begin();
ioi++; // Clearly I point to something in it_range

IteratorRange<...>::iterator it = it_range.begin(); // Implicit conversion
it++; // Do I still point to something in it_range?
```
I think that with the possibility of implicit conversion, one can silently construct an iterator that points to something that's outside of the original iterator range. In most cases I would guess that this would be unintentional.

I think that what I'd like to do is to try find out exactly where I need this conversion to take place, and ultimately try to mark this operator as explicit.

Reply:
My brain is in knots thinking about iterators over iterators. I really don't understand things because I can't seem to remember what `it` actually points to :-)